### ISSUE TITLE
When I add Pinia, execute npm run dev to report an error

### ISSUE THREAD

**Issue by estoi** [vuejs/create-vue#84]
```bash
√ Project name: ... vue-project
√ Add TypeScript? ... No / Yes
√ Add JSX Support? ... No / Yes
√ Add Vue Router for Single Page Application development? ... No / Yes
√ Add Pinia for state management? ... No / Yes
√ Add Vitest for Unit Testing? ... No / Yes
√ Add Cypress for both Unit and End-to-End testing? ... No / Yes
√ Add ESLint for code quality? ... No / Yes
```
### result
```bash
 [ERROR] [plugin vite:dep-pre-bundle] Detected inconsistent metadata for the path "node_modules/vue/dist/vue.runtime.esm-bundler.js" when it was imported here:

    node_modules/pinia/node_modules/vue-demi/lib/index.mjs:1:21:
      1 │ import * as Vue from 'vue'
        ╵                      ~~~~~

  The original metadata for that path comes from when it was imported here:

    dep:vue:2:14:
      2 │ export * from "./node_modules/vue/dist/vue.runtime.esm-bundler.js"
        ╵               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  The difference in metadata is displayed below:

   {
  -  "pluginName": null,
  +  "pluginName": "vite:dep-pre-bundle",
   }

  This is a bug in the "vite:dep-pre-bundle" plugin. Plugins provide metadata for a given path in an
  "onResolve" callback. All metadata provided for the same path must be consistent to ensure
  deterministic builds. Due to parallelism, one set of provided metadata will be randomly chosen for
  a given path, so providing inconsistent metadata for the same path can cause non-determinism.
```

**Comment by jiejiege123** [comment]
> ```shell
> √ Project name: ... vue-project
> √ Add TypeScript? ... No / Yes
> √ Add JSX Support? ... No / Yes
> √ Add Vue Router for Single Page Application development? ... No / Yes
> √ Add Pinia for state management? ... No / Yes
> √ Add Vitest for Unit Testing? ... No / Yes
> √ Add Cypress for both Unit and End-to-End testing? ... No / Yes
> √ Add ESLint for code quality? ... No / Yes
> ```
> 
> ### result
> ```shell
>  [ERROR] [plugin vite:dep-pre-bundle] Detected inconsistent metadata for the path "node_modules/vue/dist/vue.runtime.esm-bundler.js" when it was imported here:
> 
>     node_modules/pinia/node_modules/vue-demi/lib/index.mjs:1:21:
>       1 │ import * as Vue from 'vue'
>         ╵                      ~~~~~
> 
>   The original metadata for that path comes from when it was imported here:
> 
>     dep:vue:2:14:
>       2 │ export * from "./node_modules/vue/dist/vue.runtime.esm-bundler.js"
>         ╵               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 
>   The difference in metadata is displayed below:
> 
>    {
>   -  "pluginName": null,
>   +  "pluginName": "vite:dep-pre-bundle",
>    }
> 
>   This is a bug in the "vite:dep-pre-bundle" plugin. Plugins provide metadata for a given path in an
>   "onResolve" callback. All metadata provided for the same path must be consistent to ensure
>   deterministic builds. Due to parallelism, one set of provided metadata will be randomly chosen for
>   a given path, so providing inconsistent metadata for the same path can cause non-determinism.
> ```
> 
> ### It behaves normally when I don't add any additional configuration

I just had this problem. The problem I found was caused by the mismatch between the node and NPM versions, because I raised the NPM version separately before https://nodejs.org/zh-cn/download/releases/ , use the appropriate version, and the error will disappear.

我刚刚也遇到这个问题。我发现的问题是 node 和 npm 版本不匹配导致的，因为我之前单独升高了npm版本，根据https://nodejs.org/zh-cn/download/releases/，使用合适的版本，报错就消失了。

**Comment by estoi** [comment]
> 
My approach is to reduce the versions of vite, Vue router and @ vitejs / plugin Vue, and there will be no errors at present

**Comment by sodatea** [comment]
Fixed in esbuild 0.14.36
